libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -763,11 +763,11 @@ - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigati
         if (bioAuthManager.locked && bioAuthManager.hasBiometricOptedIn) {
             [bioAuthManager presentBiometricWithScene:self.view.window.windowScene];
         }
-    } else if (!self.domainUpdated && [self shouldUpdateDomain:url]) {
+    } else if ([SalesforceSDKManager sharedManager].shouldInferCustomDomain && [self shouldUpdateDomain:url]) {
         // To support case where my domain is entered through "Use Custom Domain"
-        self.domainUpdated = YES;
         decisionHandler(WKNavigationActionPolicyCancel);
         [self stopAuthentication];
+        [[SFUserAccountManager sharedInstance] setLoginHost:url.host];
```

Comment:
The host was already getting updated when the current user is set, but I added it here too so that a user can see the change in the host picker before logging in